config/validate/rules.go

```diff
@@ -82,6 +82,9 @@ func checkNodeStructure(n, g node, r *Report) {
 	case reflect.Struct:
 		for _, cn := range n.children {
 			if cg := g.Child(cn.name); cg.IsValid() {
+				if cg.field.Tag.Get("deprecated") != "" {
```

Comment:
What about providing deprecated as a string message instead of boolean so the warning can be more specific that etcd2 values don't work in the etcd section.